parsers/SG.py

```diff
@@ -267,6 +232,79 @@ def fetch_price(zone_key='SG', session=None, target_datetime=None, logger=None)
     }
 
 
+def __detect_datetime_from_solar_image(solar_image, logger):
+    w, h = solar_image.size
+    crop_left = int(w * 0.75)
+    crop_top = int(h * 0.87)
+    crop_right = int(w * 0.93)
+    crop_bottom = int(h * 0.92)
+    time_img = solar_image.crop((crop_left, crop_top, crop_right, crop_bottom))
+    processed_img = __preprocess_image_for_ocr(time_img)
+    text = image_to_string(processed_img, lang='eng', config='--psm 7 -c tessedit_char_whitelist="0123456789:- "')
```

Comment:
Similar to [my other comment](https://github.com/electricityMap/electricitymap-contrib/pull/3620/files#r774752757), [page segmentation method](https://tesseract-ocr.github.io/tessdoc/ImproveQuality#page-segmentation-method) 7 is most descriptive of the more specific image cropping that I split things into.

Here, `tessedit_char_whitelist` did improve the accuracy of datetime values detected in the image.